content/blog/angular-components-control-value-accessor/index.md

```diff
@@ -0,0 +1,419 @@
+  ) {
+    if (ngControl != null) {
+      // Setting the value accessor directly (instead of using
+      // the providers) to avoid running into a circular import.
+      ngControl.valueAccessor = this;
+    }
+  }
+    
+  // ...
+}
+```
+
+In this code sample, we're using [the `@Self` decorator](https://angular.io/api/core/Self) to tell the dependency injection system that "this component _itself_ should have been provided a `formControl` or `formControlName`". However, we want the component to work even when `FormModule` isn't being used, so we allow the dependency injection to return `null` if nothing's passed by utilizing [the `@Optional` decorator](https://angular.io/api/core/Optional).
+
+Now that you have the `ngControl`, you can access the `formControl` by using `ngControl.control`.
+
+```typescript
+ngOnInit() {
+    const control = this.ngControl && this.ngControl.control;
+    if (control) {
+        console.log("ngOnInit", control);
+        // FormControl should be available here
+    }
+}
+```
+
+You have [a ton of different props you're able to access for the control's metadata](https://angular.io/api/forms/NgControl). For example, if you want to check when errors are present, you can do the following:
+
+```typescript
+get errors() {
+	const control = this.ngControl && this.ngControl.control;
+    if (control) {
+    	return control.touched && control.errors;
+    }
+    return null;
+}
+```
+
+And then reference it in the template:
+
+```html
+<span class="inputLabel" [class.redtext]="errors">{{ placeholder }}</span>
+```
+
+Now that you have the component implementation, you can add validators to your `FormControl`:
+
+```typescript
+import { Component } from '@angular/core';
+import {FormControl, Validators} from '@angular/forms';
+
+@Component({
+  selector: 'my-app',
+  templateUrl: './app.component.html',
+  styleUrls: [ './app.component.css' ]
+})
+export class AppComponent  {
+  control = new FormControl('', Validators.required);
+}
+```
+
+<iframe src="https://stackblitz.com/edit/angular-value-accessor-dep-inject?ctl=1&embed=1&file=src/app/app.component.ts" style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
+
+
+Not only do you have [a wide range of Angular-built validators at your disposal](https://angular.io/api/forms/Validators), but you're even able to [make your own validator](https://angular.io/api/forms/Validator)!
+
+# Conclusion {#conclusion}
+
+Enabling `formControl` and `ngModel` usage is an extremely powerful tool that enables you to have feature-rich and consistent APIs across your form components. Using them, you can ensure that your consumers are provided with the functionality they'd expect in a familiar API to native elements. Hopefully, this article has provided you with more in-depth insight that you're able to take to your own components.
```

Comment:
```suggestion
Enabling `formControl` and `ngModel` usage is an extremely powerful tool that enables you to have feature-rich and consistent APIs across your form components. Using them, you can ensure that your consumers are provided with the functionality they'd expect in a familiar API to native elements. Hopefully, this article has provided you with more in-depth insight that you're able to use with your own components.
```

`to take to` sounds odd to me. If it's just me it's just me, but I wouldn't say that unless I'm looking to physically explain the difference between affect and effect.